modules/generic-scala3/src/main/scala-3/pureconfig/generic/derivation/HintsAwareConfigReaderDerivation.scala

```diff
@@ -0,0 +1,18 @@
+package pureconfig
+package generic
+package derivation
```

Comment:
nit: I'd probably prefer to keep code under `pureconfig.generic.scala3` to align with the Maven artifact name (with the exception of `semiauto` as we want this to be a drop-in replacement for `generic`).

Some of the packaging decisions within `generic` aren't consistent at the moment as back then we didn't expect there to be other generic derivation modules other than `generic` (which should have been called `generic-shapeless`), but I think we can make this one follow a more future-proof structure.